### DISCUSSION TITLE
Getting Started Mapping

### DISCUSSION THREAD

**Discussion by Fievelx** [avaraline/Avara#212] [General]
Hi all,
Is there a way to begin mapping on OS X? Or is that not possible yet? I just discovered this port a few days ago, and now that it's summer, I have time for a favorite distraction of mine (level making).
Many thanks.

**Comment by rherriman** [comment]
The short answer is yes, content creation is possible; most of the levels in the Avaraline 2021 collection(s) were created or modified using an updated process. However, it is not likely that you will enjoy it. :)

For a number of reasons (lack of resource forks in modern computing, the death of the PICT format and QuickDraw, etc.), we have elected to abandon the old file formats in favor of new, text-based, proprietary ones that are subsets of XML and JSON. You can get  a sense of what these look like by browsing the `levels` directory in the repo. Here's an example: https://github.com/avaraline/Avara/tree/main/levels/avaraline-2021-strict-mode

The plan is to eventually have a mature editor for working with these files. Work began on the editor (https://github.com/avaraline/otto) but it hasn't been touched in some time, and has a long ways to go. It is useful, at least, for providing a live preview of changes. You can play with a version of it here: https://vastan.net/dump/otto/ (@assertivist is this up to date?)

Unfortunately, the port also currently bundles the levels as part of the build process, which means that some familiarity with building the game is somewhat necessary to create or edit levels. We want to eventually decouple the game content from the executable in some fashion, and provide some sort of network-based asset distribution system, but we haven't settled on a plan for that yet.

Obviously this is a very big topic. If you're still interested in trying your hand at it, I'd recommend swinging by the Avara Discord (https://discord.gg/gMMF4wXv) to discuss it with us there!

**Reply by rherriman** [reply]
I suppose I should also mention that while we have new level file formats, we also have utilities for converting old level sets into the new formats! So it's also possible to, say, create an OS9 emulation environment with ClarisWorks, ResEdit, and the original Avara, create levels within that environment, transfer those files into OSX and run them through the level converter. Someone on the Discord could potentially help you with using the converter as well. We're a friendly bunch!

**Reply by Fievelx** [reply]
Thanks for the comprehensive reply.
It does seem to me that you've already got all the goat levels. The 2021 set is really good and novel. The only thing I feel that might be needed are some really good solo/co-op missions.
I spent today making a couple simple svg mockups. I can send them, if anyone is curious. I'd like some layout advice. They might be pretty lame. 
Otto is great for learning variables, but I can't figure out how to make something new.
I'll be joining discord soon.
